source/Classroom/Troubleshooting/Authentication/list_upload_troubleshooting.md

```diff
@@ -1,53 +1,100 @@
+{% anchor h3 %} I have more than one of the same email in my database {% anchor h3 %}
 
 With SendGrid’s Marketing Campaigns feature, the unique identifier is the email address. So, if you upload the same email address multiple times, the custom field data associated with that email will be updated with each upload to the most recently uploaded/updated information. This feature helps prevent you from accidentally emailing the user after they have unsubscribed.
 
 What you may want to do is add logic to set a custom field, based on the reason why you have duplicate emails in your system (for example multiple product lines) and then segment your user to be in lists based on those custom fields.
 
 **‹ ** [Custom Fields](https://sendgrid.com/docs/User_Guide/Marketing_Campaigns/custom_fields.html)
 
+
+{% anchor h2 %} Specific Upload Errors {% endanchor h2 %}
+
+{% anchor h3 %} We were unable to detect an email column in the CSV file headers. The email column must always have a header of "email" {% endanchor h3 %}
+
+- If there is a header labeled “email”, try moving it to the column A, so it is the first header in your list.
+- Your CSV _must_ include an email header.
+
+![]({{root_url}}/images/listupload_1.png)
+
+{% anchor h3 %} Some of your custom fields have not been selected {% endanchor h3 %}
+
+- If you have worked with your CSV multiple times, or have made some adjustments to the data, there can be some hidden characters in the CSV file.
+- A common fix for this error would be to select all and copy your data to a new CSV spreadsheet. This should remove any hidden characters.
+
+![]({{root_url}}/images/listupload_2.png)
+
+{% anchor h3 %} Each custom field can only be applied to a column once. Please check your columns and try again {% endanchor h3 %}
+
+- Check your CSV for duplicate fields, especially fields that have matching information as your “email”, “first_name” and “last_name” fields. These are reserved field names, and cannot be edited.
+- For example, if you have a “Name” field, it may conflict with your “first_name” reserved field.
+
+![]({{root_url}}/images/listupload_3.png)
+
+{% anchor h3 %} We were unable to detect any headers in your CSV file {% endanchor h3 %}
+
+- Please be sure to include headers at the top of each column. The “email” header is required for all CSV uploads. Header fields can be blank in your CSV, but must be defined later during the upload process.
+
+![]({{root_url}}/images/listupload_4.png)
+
+{% anchor h3 %} Float type conversion error {% endanchor h3 %}
+
+- Dashes or decimal places past the hundredth place need to be categorized as text_fields (-100, 123.123 are text fields, 100, 12.12 can be number fields).
+Number fields can include monetary values without the currency symbols. For example, 19.95 would be a number field, and $19.95 including the ($) would be a text field.
+- As you are uploading your CSV, make sure you select the correct fields for your Field type.
+
```

Comment:
Aw crap, is the first part of this an error message? or is that what's in the h3 tag?